snapcraft/tests/fixture_setup.py

```diff
@@ -29,16 +33,94 @@ def setUp(self):
         os.chdir(self.path)
 
 
-class StagingStore(fixtures.Fixture):
+class _FakeServerRunning(fixtures.Fixture):
+
+    # To be defined by child fixtures.
+    fake_server = None
+
+    def setUp(self):
+        super().setUp()
+        self._start_fake_server()
+
+    def _start_fake_server(self):
+        server_address = ('', 0)
+        self.server = self.fake_server(server_address)
+        server_thread = threading.Thread(target=self.server.serve_forever)
+        server_thread.start()
+        self.addCleanup(self._stop_fake_server, server_thread)
+        self.url = 'http://localhost:{}/'.format(self.server.server_port)
+
+    def _stop_fake_server(self, thread):
+        self.server.shutdown()
+        self.server.socket.close()
+        thread.join()
```

Comment:
This can hang if the client didn't close its connection (i.e. it's not enough to consume the response).